From: Robert Buchholz <rbu@gentoo.org>
To: lcdproc@lists.omnipotent.net
Subject: Re: [Lcdproc] [ANNOUNCE] lcd-stuff 0.1.3
Cc: Bernhard Walle <bernhard.walle@gmx.de>
Date: Wed, 8 Aug 2007 22:06:33 +0200

When compiling against the latest version of libmrss
(0.18.0), compilation breaks because mrss.h includes
sys/socket.h. That header uses SHUT_RDWR as a variable name
(which has to be undefined), but lcd-stuff's shared/sockets.h
defines that macro.

The attached patch fixes this on lcd-stuff's side.

Index: lcd-stuff-0.1.3/src/rss.c
===================================================================
--- lcd-stuff-0.1.3.orig/src/rss.c
+++ lcd-stuff-0.1.3/src/rss.c
@@ -24,12 +24,12 @@
 #include <string.h>
 #include <errno.h>
 
+#include <mrss.h>
+
 #include <shared/report.h>
 #include <shared/sockets.h>
 #include <shared/str.h>
 
-#include <mrss.h>
-
 #include "rss.h"
 #include "main.h"
 #include "constants.h"
